shiny/ui/_theme_brand.py

```diff
@@ -0,0 +1,528 @@
+                f"{preset!r} is not a valid Bootstrap preset provided by Shiny. "
+                f"Valid presets are {shiny_theme_presets}."
+            )
+
+        self.version = v_major
+        self.preset: ShinyThemePreset = preset
+        self.defaults = kwargs
+
+    @classmethod
+    def from_brand(cls, brand: Brand):
+        defaults: dict[str, str | int | bool | float | None] = {}
+
+        if brand.defaults:
+            if brand.defaults and "bootstrap" in brand.defaults:
+                if isinstance(brand.defaults["bootstrap"], dict):
+                    brand_defaults_bs: dict[str, str] = brand.defaults["bootstrap"]
+                    defaults.update(brand_defaults_bs)
+            if "shiny" in brand.defaults and "theme" in brand.defaults["shiny"]:
+                if isinstance(brand.defaults["shiny"]["theme"], dict):
+                    # TODO: Use brand.defaults.shiny.theme.defaults instead
+                    # TODO: Validate that it's really a dict[str, scalar]
+                    brand_shiny_theme: dict[str, str] = brand.defaults["shiny"]["theme"]
+                    defaults.update(brand_shiny_theme)
+
+                # TODO: Get functions, mixins, rules as well
+
+        return cls(**defaults)
+
+
+class ThemeBrand(Theme):
+    def __init__(
+        self,
+        brand: Brand,
+        *,
+        include_paths: Optional[str | Path | list[str | Path]] = None,
+    ):
+
+        name = self._get_theme_name(brand)
+        brand_bootstrap = BrandBootstrapConfig.from_brand(brand)
+
+        # Initialize theme ------------------------------------------------------------
+        super().__init__(
+            name=name,
+            preset=brand_bootstrap.preset,
+            include_paths=include_paths,
+        )
+
+        self.brand = brand
+
+        # Prep Sass and CSS Variables -------------------------------------------------
+        sass_vars_colors, css_vars_colors = self._prepare_color_vars()
+        sass_vars_typography = self._prepare_typography_vars()
+
+        # Theme -----------------------------------------------------------------------
+        # Defaults are added in reverse order, so each chunk appears above the next
+        # layer of defaults. The intended order in the final output is:
+        # 1. Brand Sass color and typography vars
+        # 2. Brand's Bootstrap Sass vars
+        # 3. Gray scale variables from Brand fg/bg or black/white
+        # 4. Fallback vars needed by additional Brand rules
+
+        self.add_defaults("", "// *---- brand: end of defaults ----* //", "")
+        self._add_sass_ensure_variables()
+        self._add_sass_brand_grays()
+        self._add_defaults_brand_bootstrap(brand_bootstrap)
+        self._add_defaults_typography(sass_vars_typography)
+        self._add_defaults_color(sass_vars_colors)
+
+        # Brand rules (now in forwards order)
+        self._add_rules_brand_colors(css_vars_colors)
+        self._add_sass_brand_rules()
```

Comment:
Consider making most/all of these `@staticmethod`s, which would have 2 benefits: (1) it makes it more clear from the `__init__` what side-effects you are performing to theme and (2) the objects you pass into those side-effect actions are more easily inspectable

Reply:
I made the Sass var helpers static methods in dbc475960a08f28d8861ed8eb48660a5ecab5246

I'm not sure how the theme-modifying methods could be made static (they have to modify self), but maybe you want them to be static methods that take a `theme` argument?